packages/fast-tooling-react/src/form/utilities/form.tsx

```diff
@@ -598,13 +599,27 @@ export function getErrorFromDataLocation(
                     ) {
                         error = validationError.message;
                     }
-                } else if (
-                    validationError.dataPath.slice(
-                        0,
-                        matchingDataLocationToDataPath.length
-                    ) === matchingDataLocationToDataPath
-                ) {
-                    error = "Contains invalid data";
+                } else {
+                    const dataLocationItems: string[] = `.${normalizeDataLocationToDotNotation(
+                        validationError.dataPath
+                    )}`.split(".");
+                    let containsInvalidData: boolean = false;
```

Comment:
Array's have a `some` method to serve exactly this case: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some